Vwitter/View Controllers/ComposeViewController.m

```diff
@@ -57,8 +110,31 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
         
         SelectAudienceViewController *selectAudienceVC = [segue destinationViewController];
         selectAudienceVC.ventContent = self.ventContent.text;
+        selectAudienceVC.selectedTrack = self.selectedTrack;
         
     }
+    else if ([segue.identifier isEqualToString:@"songPickerSegue"]) {
+        SongPickerViewController *songPickerVC = CAST_TO_CLASS_OR_NIL([segue destinationViewController], SongPickerViewController);
+        songPickerVC.delegate = self;
+    }
+}
+
+- (void)setSongTitleLabelX:(CGFloat)x {
+    self.songTitleLabel.frame = CGRectMake(
+        x, // x
+        self.songTitleLabel.frame.origin.y, // y
+        self.songTitleLabel.frame.size.width, // width
+        self.songTitleLabel.frame.size.height // height
+    );
+    
+}
+
+- (void)passSelectedTrack:(SpotifyTrack *)selectedTrack {
+    self.selectedTrack = selectedTrack;
+    self.songTitleLabel.text = selectedTrack.trackName;
```

Comment:
I still really think you should use an NSAttributed string here with both artist + song name :)